experimental/saiexperimentaldashha.h

```diff
@@ -0,0 +1,442 @@
+ *
+ * @param[in] dash_ha_id HA object id
+ * @param[in] number_of_counters Number of counters in the array
+ * @param[in] counter_ids Specifies the array of counter ids
+ * @param[out] counters Array of resulting counter values.
+ *
+ * @return #SAI_STATUS_SUCCESS on success, failure status code on error
+ */
+typedef sai_status_t (*sai_get_dash_ha_stats_fn)(
+        _In_ sai_object_id_t dash_ha_id,
+        _In_ uint32_t number_of_counters,
+        _In_ const sai_stat_id_t *counter_ids,
+        _Out_ uint64_t *counters);
+
+/**
+ * @brief Get DASH HA statistics counters extended.
+ *
+ * @param[in] dash_ha_id HA object id
+ * @param[in] number_of_counters Number of counters in the array
+ * @param[in] counter_ids Specifies the array of counter ids
+ * @param[in] mode Statistics mode
+ * @param[out] counters Array of resulting counter values.
+ *
+ * @return #SAI_STATUS_SUCCESS on success, failure status code on error
+ */
+typedef sai_status_t (*sai_get_dash_ha_stats_ext_fn)(
+        _In_ sai_object_id_t dash_ha_id,
+        _In_ uint32_t number_of_counters,
+        _In_ const sai_stat_id_t *counter_ids,
+        _In_ sai_stats_mode_t mode,
+        _Out_ uint64_t *counters);
+
+/**
+ * @brief Clear DASH HA statistics counters.
+ *
+ * @param[in] dash_ha_id HA object id
+ * @param[in] number_of_counters Number of counters in the array
+ * @param[in] counter_ids Specifies the array of counter ids
+ *
+ * @return #SAI_STATUS_SUCCESS on success, failure status code on error
+ */
+typedef sai_status_t (*sai_clear_dash_ha_stats_fn)(
+        _In_ sai_object_id_t dash_ha_id,
+        _In_ uint32_t number_of_counters,
+        _In_ const sai_stat_id_t *counter_ids);
+
+/**
+ * @brief Clear all DASH HA statistics counters.
+ *
+ * @param[in] dash_ha_id HA object id
+ *
+ * @return #SAI_STATUS_SUCCESS on success, failure status code on error
+ */
+typedef sai_status_t (*sai_clear_dash_ha_all_stats_fn)(
+        _In_ sai_object_id_t dash_ha_id);
+
+/**
+ * @brief DASH HA FSM notifications
+ *
+ * @count data[count]
+ *
+ * @param[in] count Number of notifications
+ * @param[in] data Pointer to DASH HA FSM event notification data array
+ */
+typedef void (*sai_dash_ha_fsm_event_notification_fn)(
+        _In_ uint32_t count,
+        _In_ const sai_dash_ha_fsm_event_notification_data_t *data);
+
+typedef struct _sai_dash_ha_api_t
+{
+    sai_create_dash_ha_fn                      create_dash_ha;
+    sai_remove_dash_ha_fn                      remove_dash_ha;
+    sai_set_dash_ha_attribute_fn               set_dash_ha_attribute;
+    sai_get_dash_ha_attribute_fn               get_dash_ha_attribute;
+    sai_get_dash_ha_stats_fn                   get_dash_ha_stats;
+    sai_get_dash_ha_stats_ext_fn               get_dash_ha_stats_ext;
+    sai_clear_dash_ha_stats_fn                 clear_dash_ha_stats;
+    sai_clear_dash_ha_all_stats_fn             clear_dash_ha_all_stats;
+    sai_start_dash_ha_fsm_fn                   start_dash_ha_fsm;
```

Comment:
these APIs will be non-standard APIs in SAI API, and it will be better to avoid these changes. otherwise, to support it there will be a lot of additional changes in sairedis, vendor sai, client/server sai, syncd and so on.

and as an example: start_dash_ha_fsm can be an sai set API call to update the oper_role or other attribuet on ha fsm.

Reply:
same for stop, activate, switchover.

for doing flow resimulation, we will provide the  flow resimulation api in separate dedicated doc, so we don't need it here.

Reply:
These triggers are per vip-id in the API model.